# **2022 NFL Combine - Project 2**
### Analyzing the linear relationship between two quantitative variables.

# **Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that add extra features to the basic setup. The name of the modules is after the `import` statement, and the purpose is in a non-code comment after thew hashtag (#).



In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image
import statsmodels.api as sm


In [ ]:
# Assigns the URL of the image to display to the name 'image_url'.
image_url = 'https://as2.ftcdn.net/v2/jpg/04/31/74/37/1000_F_431743763_in9BVVzCI36X304StR89pnxyUYzj1dwa.jpg'

# Display the image
Image(url=image_url, width = 575)

# **Context**

National Invitational Camp (NIC), more commonly known as the NFL Scouting Combine, began in 1982 when National Football Scouting, Inc. first conducted a camp for its member NFL clubs in Tampa, Florida. The key purpose then, same as it is today, was to ascertain medical information on the top draft eligible prospects in college football. The inaugural NIC was attended by a total of 163 players and established a foundation for future expansion.

As football and the art of evaluating players has evolved, so has the NFL Scouting Combine. While medical examinations remain the number one priority of the event, athletes will also participate in a variety of psychological and physical tests, as well as, formal and informal interviews with top executives, coaches and scouts from all 32 NFL teams. NIC is the ultimate four day job interview for the top college football players eligible for the upcoming NFL Draft.

Attribution: NFLCombine.net


# **About the Dataset**

This dataset contains 133 rows corresponding to a random sample of drafted players. A total of 9 variables are provided as listed below:

| Variable Name(s)      | Description                            |
|:----------------------|:--------------------------------------|
| Player                | Player ID, which is the player's name  |
| Pos                   | Position the player plays              |
| School                | College the player attended            |
| Ht                    | Player height (inches)                 |
| Wt                    | Player weight (lbs)                    |
| 40yd                  | Time to run the 40-yard dash (seconds) |
| Vertical              | Vertical jump height (inches)          |
| Broad Jump            | Horizontal distance covered (inches) (aka long jump) |
| Drafted (tm/rnd/yr)   | Team, round, and year the player was drafted |



Let's take a look at the data. To do this, first we import it directly from the url below.



# **A Snippet of the Data**

In [ ]:
# Assigns the URL where the data file is stored to 'file_path'.
url='https://raw.githubusercontent.com/thamilton562/STAT108_Projects_Students/main/DataSets/NFLCombine.csv'

# Reads in the CSV data file and assigns it to the DataFrame 'df'.
df=pd.read_csv(url)

Next, we can display the data by *typing the name* of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)

# When you type the object name, the object gets printed.
df

Player   Pos       School  Ht   Wt  40yd  Vertical  Broad Jump  \
0      Myjai Sanders  EDGE   Cincinnati  77  228  4.67      33.0         120   
1    Keaontay Ingram    RB          USC  73  221  4.53      34.5         122   
2       Jesse Luketa    LB     Penn St.  75  253  4.89      37.5         114   
3      Marquis Hayes    OG     Oklahoma  77  318  5.30      23.5         102   
4      Troy Andersen    LB  Montana St.  76  243  4.42      36.0         128   
..               ...   ...          ...  ..  ...   ...       ...         ...   
128     Jahan Dotson    WR     Penn St.  71  178  4.43      36.0         121   
129   Brian Robinson    RB      Alabama  74  225  4.53      30.0         119   
130     Percy Butler     S    Louisiana  73  194  4.36      31.5         123   
131      Cole Turner    TE       Nevada  78  246  4.76      27.0         120   
132       Chris Paul    OG        Tulsa  76  323  4.89      27.0         109   

                                 Drafted (tm/rnd/yr)  
0        Arizona Cardinals / 3rd / 100th pick / 2022  
1        Arizona Cardinals / 6th / 201st pick / 2022  
2        Arizona Cardinals / 7th / 256th pick / 2022  
3        Arizona Cardinals / 7th / 257th pick / 2022  
4           Atlanta Falcons / 2nd / 58th pick / 2022  
..                                               ...  
128   Washington Commanders / 1st / 16th pick / 2022  
129   Washington Commanders / 3rd / 98th pick / 2022  
130  Washington Commanders / 4th / 113th pick / 2022  
131  Washington Commanders / 5th / 149th pick / 2022  
132  Washington Commanders / 7th / 230th pick / 2022  

[133 rows x 9 columns]

# **INSTRUCTIONS**

* Use Python to analyze the data set and complete each of the following.
* Replace ellipsis (...) with the relavent names or code.  
* For problems that require a written response, replace the ellipsis (...) by double clicking the text box to start typing.
* Reference the tutorial from activity for assistance.
* Attend office hours if you still need help.

This assignment is intended to explore the univariate linear relationships between quantitative variables in the data. Choose the dependent and independent variables based on your last name.

| Last Name | Dependent Variable (y) | Independent Variable (x) |
|-----------|------------------------|--------------------------|
| A-L       | Weight                 | Height                   |
| M-Z       | Vertical jump height   | Broad jump distance      |




Based on your last name, which variable is explanatory and which is the response?

Explanatory: ...

Response: ...

What do you think the direction (positive/negative) and strength (weak/moderate/strong) will be?

# ...

# **QUESTION 1**

## Scatterplot

**1.1)** Construct a scatterplot that could be used to show the relationship between the variables stated above.

In [ ]:
# Create a scatter plot using Plotly Express

# Scatter plot for A-L
# STUDENTS: replace ... as stated
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Ht',
                 # Replace ... with the response variable name
                 y='Wt',
                         # Replace ... with a better x-axis label
                 labels={'Ht': 'Height',
                         # Replace ... with a better y-axis label
                         'Wt': 'Weight'})

# Updating layout:
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot:
scatter_plot_AL.show()

# Scatter plot for M-Z
# Create a scatter plot using Plotly Express
scatter_plot_MZ = px.scatter(df,
                 x='Broad Jump', #explanatory variable name
                 y='Vertical', #response variable name
                 labels={'Broad Jump': 'Broad Jump Distance', #updates axis labels
                         'Vertical': 'Vertical Jump Height'})

# Updating layout:
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=575,                                 # Sets the width of the plot.
    height=400                                 # Sets the height of the plot for portrait mode.
)

# Show the plot
scatter_plot_MZ.show()

**1.2)** Describe the relationship between the two variables and include context, and identify the direction, strength, and form. Since you don't yet know the value of correlation (r), estimate what you think the strength is (weak/moderate/strong).



**A-L:** There is a moderate, positive, linear relationship between height and weight.

**M-Z:** There is a strong, positive, linear relationship between broad jump distance and vertical jump height.

**1.3)** Is this what you expected?


# ...

# **Question 2**
## Correlation Coefficient

**2.1)** Calculate the value of the correlation coefficient.

In [ ]:
# Calculate the correlation coefficient

# STUDENTS: Replace ... as stated
# Replace the 1st ... with the response variable name.
# Replace the 2nd ... with the explanatory variable name
correlation_AL = df['Wt'].corr(df['Ht'])
correlation_MZ = df['Vertical'].corr(df['Broad Jump'])

# Print the correlation coefficient
print(f"A-L: Correlation coefficient: r = {correlation_AL}")
print(f"\nM-Z: Correlation coefficient: r = {correlation_MZ}")

A-L: Correlation coefficient: r = 0.761738701303471

M-Z: Correlation coefficient: r = 0.8561549400599138


**2.2)** Interpret the correlation coefficient.



**A-L:** There is a moderate, positive, linear relationship between height and weight.

**M-Z:** There is a strong, positive, linear relationship between broad jump distance and vertical jump height.

**2.3)** Now that you know the value of correlation, r, did the strength match what you wrote in 1.2?

# ...

# **Question 3**

## Least Squares Regression

**3.1)** Calculate the linear model and add the regression line to the scatterplot.


In [ ]:
# Calculate the LSRL and add the regression line to the scatterplot.

# A-L
# Extract the variables from the dataframe
# STUDENTS: Replace ... as stated below.
# Replace ... with the explantory variable name
X = df['Ht']
# Replace ... with the response variable name
Y = df['Wt']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
# STUDENTS: Do not change anything in this block of code.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
# STUDENTS: Do not change anything in this block of code.
model_AL = sm.OLS(Y, X_with_const).fit()

# STUDENTS: Replace ... as stated below.
scatter_plot_AL = px.scatter(df,
                 # Replace ... with the explanatory variable name
                 x='Ht',
                 # Replace ... with the response variable name
                 y='Wt',
                         # Replace ... with a better x-axis label
                 labels={'Ht': 'Height',
                         # Replace ... with a better y-axis label
                         'Wt': 'Weight'})

# Updating layout
# STUDENTS: Do not change anything in this block of code.
scatter_plot_AL.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot A-L',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_AL.add_scatter(x=X, y=model_AL.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_AL.show()

# M-Z
# Extract the variables from the dataframe
X = df['Broad Jump']
Y = df['Vertical']

# Add a constant to the explanatory variable for the regression model.
# This forces Statsmodels to calculate the y-intercept.
X_with_const = sm.add_constant(X)

# Fit the linear regression model (calculate slope and y-intercept)
model_MZ = sm.OLS(Y, X_with_const).fit()

# If we repeatedly run this section of code then a new regression line is
# added each run, unless we recreate the scatterplot each iteration.
scatter_plot_MZ = px.scatter(df,
                 x='Broad Jump', #explanatory variable name
                 y='Vertical', #response variable name
                 labels={'Broad Jump': 'Broad Jump Distance', #updates axis labels
                         'Vertical': 'Vertical Jump Height'})

# Updating layout: Students do not change anything in this block of code.
scatter_plot_MZ.update_layout(
    plot_bgcolor='rgba(255,255,255,1)',        # Sets the background color of the plot area to white with full opacity.
    xaxis=dict(
        showline=True,                         # Displays a line on the x-axis.
        showgrid=False,                        # Hides the grid lines on the x-axis.
        linecolor='black'                      # Sets the color of the x-axis line to black.
    ),
    yaxis=dict(
        showline=True,                         # Displays a line on the y-axis.
        showgrid=False,                        # Hides the grid lines on the y-axis.
        linecolor='black'                      # Sets the color of the y-axis line to black.
    ),
    title={
        'text': 'Scatter Plot M-Z',   # Sets the title text.
        'y': 0.9,                              # Positions the title 90% of the way up the plot.
        'x': 0.5,                              # Centers the title horizontally.
        'xanchor': 'center',                   # Anchors the title at its center on the x-axis.
        'yanchor': 'top',                      # Anchors the title at the top on the y-axis.
        'font': dict(
            size=16                            # Sets the title font size to 16 (smaller than default).
        ),
    },
    width=600,                                 # Sets the width of the plot.
    height=400,                                # Sets the height of the plot for portrait mode.
    showlegend=False                           # Disables the legend (key) display
)

# Add the regression line to the scatterplot
scatter_plot_MZ.add_scatter(x=X, y=model_MZ.predict(X_with_const),
                         mode='lines', name='Regression Line')

# Show the plot
scatter_plot_MZ.show()

**3.2)** Print the coefficients (Intercept and Slope)



In [ ]:
# Print the coefficients (Intercept and Slope)

# STUDENTS: Do not change the code below.
# The coefficients were calculated when the regression line was added to the
# scatterplot and stored in 'model'. This assigns the values of the y-intercept
# and slope to 'coefficients', then prints the values.

# A-L
coefficients_AL = model_AL.params
coefficients_AL.name='Coefficients'
coefficients_AL

# M-Z
coefficients_MZ = model_MZ.params
coefficients_MZ.name='Coefficients'
coefficients_MZ

# Output the DataFrames
print("A-L Group:")
print(coefficients_AL)
print("\nM-Z Group:")
print(coefficients_MZ)

A-L Group:
const   -830.926682
Ht        14.473534
Name: Coefficients, dtype: float64

M-Z Group:
const        -20.192984
Broad Jump     0.446065
Name: Coefficients, dtype: float64


**3.3)** Write the equation of the line that predicts your assigned response variable from your assigned explanatory variable. To type 𝑦̂ you can type “y-hat” or “predicted y”. Do NOT round.



**A-L:** y-hat = -830.926682 + 14.473534x

**M-Z:** y-hat = -20.192984 + 0.446065x

# **Question 4**
## Slope
Interpret the slope


**A-L:** We expect weight to increase 14.473534 lbs when height increases by 1 inch.

**M-Z:** We expect the vertical jump height to increase by 0.446065 inches when the broad jump distance increases by 1 inch.

# **QUESTION 5**
## Y-Intercept

Interpret the y-intercept of the least squares regression line in the context of this study. State whether the interpretation is reasonable.

**A-L:** We expect the weight to be -830.926682 lbs when the height is 0 inches.

No, this does not make sense. (No explanation required)

**M-Z:** We expect the vertical jump height to be -20.192984 inches when the broad jump distance is 0 inches.

No, this does not make sense. (No explanation required)

# **Question 6**
## Prediction

Rachaad White, a running back from Arizona State University, was at the combine. Predict the value of your dependent variable for Rachaad White using the actual value of the independent variable shown in the table below. Type your work below. Round to two (2) decimal places.

| Last Name      | Independent Variable (x)                |
|----------------|----------------------------------------|
| **A-L**        | Height = 73 inches |
| **M-Z**        | Broad Jump = 125 inches |


**A-L:** y-hat = -830.9266816295979 + 14.47353431272527(73) = 225.64 pounds

**M-Z:** y-hat = -20.192983769315266 + 0.4460653838208126(0125) = 35.57 inches

In [ ]:
# This is NOT in the student notebook.
# A-L prediction
intercept_AL = model_AL.params['const']
slope_AL = model_AL.params['Ht']
predicted_AL = (intercept_AL + slope_AL * 73)
print(f"A-L: y-hat = {intercept_AL} + {slope_AL}(73) = {predicted_AL}")

# M-Z prediction
intercept_MZ = model_MZ.params['const']
slope_MZ = model_MZ.params['Broad Jump']
predicted_MZ = (intercept_MZ + slope_MZ * 125)
print(f"M-Z: y-hat = {intercept_MZ} + {slope_MZ}(0125) = {predicted_MZ}")

A-L: y-hat = -830.9266816295979 + 14.47353431272527(73) = 225.64132319934697
M-Z: y-hat = -20.192983769315266 + 0.4460653838208126(0125) = 35.56518920828631


# **Question 7**
## Compare
**7.1)** Look up the actual value of your dependent variable for the value of the independent variable above. We know it is for Criminology majors, but be sure to type in the value of the independent variable listed in Question 6 in the code below.

In [ ]:
# Filter the datafram to find the row where your independent variable
#   equals the value from the table in Question 6.

#A-L
# STUDENTS:
# Replace the 1st ... with the explantory variable name
# Replace the 2nd ... with the value from the table in Question 6
# Replace the 3rd ... with the response variable name
actual_dependent_AL = df[df['Ht']== 73]['Wt']
actual_AL = int(actual_dependent_AL.values[0])
print(f"A-L: The actual value of the dependent variable is: {actual_AL}")

#M-Z
actual_dependent_MZ = df[df['Broad Jump']== 125]['Vertical']
actual_MZ = int(actual_dependent_MZ.values[0])
print(f"M-Z: The actual value of the dependent variable is: {actual_MZ}")

A-L: The actual value of the dependent variable is: 221
M-Z: The actual value of the dependent variable is: 34


**7.2)** Compare the predicted value of your dependent variable from Question 6 to the actual value from 7.1. Use the actual name of your dependent variable, as modeled in activity.


**A-L:** The actual weight is lower than the predicted value.

**M-Z:** The actual vertical jump distance is shorter than the predicted value.

# **QUESTION 8**

Generate a paragraph of at least 100 words to address one of the following questions. That is, answer only 8a or 8b, but not both.

**8a)** Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

...

--OR--

**8b)** Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

...


<br><br>
### Once you are done and ready to submit, follow the instructions below to save as a PDF and submit to GradeScope.

### Save as PDF
Note 1: You do not have to select Print Preview. You can print directly from the notebook.
Note 2: Image and graph sizes have been set so you should be able to see them correctly without making any changes to the browser width or the layout (portrait vs landscape).
1. Run all code one last time and make sure your graphs can be seen.
2. File -> Print (or ctrl-p/cmnd-p)
3. Change the "Desination" to PDF.
4. Save the PDF, taking note of where it is saved.

### Submit to GradeScope
**Watch the "GradeScope Submission" video for help.**
1. Login to the Canvas course
2. Click on GradeScope in the course navigation.
3. If you see multiple courses in GradeScope, click on the STAT 108 course
4. Click on the name of the assignment that matches your data set
5. Click on "Submit Work", select PDF
6. Select the PDF you just created
7. You need to tell GradeScope which page each problem answer/output is on. You should see a list of problems on the left, and a display of pages (thumbnails) on the right. Assign pages to questions by clicking on the question number on the left, then clicking on all pages that question is on.
8. After ALL questions have been assigned to their respective page(s), click "Submit"

#### **Still need help? Your STAT 108 team is here to help. Take your laptop to office hours.**
